pysite/views/api/bot/doc.py

```diff
@@ -0,0 +1,110 @@
+from flask import jsonify
+from schema import Optional, Schema
+
+from pysite.base_route import APIView
+from pysite.constants import ValidationTypes
+from pysite.decorators import api_key, api_params
+from pysite.mixins import DBMixin
+
+
+GET_SCHEMA = Schema([
+    {
+        Optional("package"): str
+    }
+])
+
+POST_SCHEMA = Schema([
+    {
+        "package": str,
+        "base_url": str,
+        "inventory_url": str
+    }
+])
+
+DELETE_SCHEMA = Schema([
+    {
+        "package": str
+    }
+])
+
+
+class DocView(APIView, DBMixin):
+    path = "/docs"
+    name = "api.bot.docs"
+    table_name = "pydoc_links"
+
+    @api_key
+    @api_params(schema=GET_SCHEMA, validation_type=ValidationTypes.params)
+    def get(self, params=None):
+        """
+        Fetches documentation metadata from the database.
+
+        - If `package` is provided, fetch metadata
+        for the given package, or `{}` if not found.
+
+        - If `package` is not provided, return all
+        packages known to the database.
+
+        Data must be provided as params.
+        API key must be provided as header.
+        """
+
+        if params:
+            package = params[0].get("package")
+            data = self.db.get(self.table_name, package) or {}
+        else:
+            data = self.db.pluck(self.table_name, ("package", "base_url", "inventory_url")) or []
+
+        return jsonify(data)
+
+    @api_key
+    @api_params(schema=POST_SCHEMA, validation_type=ValidationTypes.json)
+    def post(self, json_data):
+        """
+        Adds a new documentation metadata object.
+
+        If the `package` passed in the data
+        already exists, it will be updated instead.
+
+        Data must be provided as JSON.
+        API key must be provided as header.
+        """
+
+        json_data = json_data[0]
+
+        package = json_data["package"]
+        base_url = json_data["base_url"]
+        inventory_url = json_data["inventory_url"]
+
+        self.db.insert(
+            self.table_name,
+            {
+                "package": package,
+                "base_url": base_url,
+                "inventory_url": inventory_url
+            },
+            conflict="update"
+        )
+
+        return jsonify({"success": True})
+
+    @api_key
+    @api_params(schema=DELETE_SCHEMA, validation_type=ValidationTypes.json)
+    def delete(self, json_data):
+        """
+        Deletes a documentation metadata object.
+        Expects the `package` to be deleted to
+        be specified as a request parameter.
+
+        Data must be provided as params.
+        API key must be provided as header.
+        """
+
+        package = json_data[0]["package"]
```

Comment:
This schema setup is intended to allow one to specify multiple sets of parameters. You should support this in your endpoints.